C++/source/dynamic_programming/longest_increasing_subsequence.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+   Longest Increasing Subsequence
+   ------------------------------
+   Given an unsorted array of integers, return the longest increasing subsequence
+
+   Time complexity
+   --------------
+   O(N), where N is size of array
+
+   Space complexity
+   ---------------
+   O(N), where N is size of array
+ */
+
+#include <iostream>
+#include <vector>
+#include <climits>
+#include <algorithm>
+
+using namespace std;
+
+int find_prev_smaller(const vector<int> &a, int x) {
```

Comment:
Add a comment that briefly describes what this function does.
Also, consider a better name for `a` (`values`? `input`?) and `x` (`current_index`?).